### PR TITLE
Add support for serializing to a proto stream.

### PR DESCRIPTION
**Pull request by wohe**
Also changes the ground truth tools to read from such a file.

This resolves #335. Related to #253.

### CODE REVIEW

**Comment by wohe** [comment]
This is my suggestion where to add serialization support.

/cc @brandon-northcutt
/cc @jihoonl

File: cartographer/mapping/map_builder.cc [hunk 1]
```diff
@@ -132,6 +132,14 @@ string MapBuilder::SubmapToProto(const mapping::SubmapId& submap_id,
   return "";
 }
 
+void MapBuilder::SerializeState(io::ProtoStreamWriter* writer) {
```
Comment (damonkohler) [inline cartographer/mapping/map_builder.cc]: const

File: cartographer/ground_truth/autogenerate_ground_truth_main.cc [hunk 2]
```diff
@@ -20,14 +20,16 @@
...
+#include "cartographer/io/proto_stream.h"
 #include "cartographer/mapping/proto/sparse_pose_graph.pb.h"
 #include "cartographer/transform/transform.h"
 #include "gflags/gflags.h"
 #include "glog/logging.h"
 
 DEFINE_string(
     pose_graph_filename, "",
-    "File with the pose graph proto from which to generate ground truth data.");
+    "File with the proto stream containing the pose graph from which to "
```
Comment (damonkohler) [inline cartographer/ground_truth/autogenerate_ground_truth_main.cc]: Proto stream file containing the pose graph used to generate ground truth.

File: cartographer/ground_truth/compute_relations_metrics_main.cc [hunk 3]
```diff
@@ -36,7 +37,8 @@
 
 DEFINE_string(
     pose_graph_filename, "",
-    "File with the pose graph proto from which to assess the quality.");
+    "File with the proto stream containing the pose graph from which to "
```
Comment (damonkohler) [inline cartographer/ground_truth/compute_relations_metrics_main.cc]: Proto stream file containing the pose graph used to assess quality.

**Pull request merged**